crates/nu-parser/src/parser.rs

```diff
@@ -3646,8 +3646,19 @@ pub fn parse_signature_helper(working_set: &mut StateWorkingSet, span: Span) ->
                                                 }
                                             }
                                         }
+
+                                        *default_value = if let Ok(constant) =
+                                            eval_constant(working_set, &expression)
+                                        {
+                                            Some(constant)
+                                        } else {
+                                            working_set.error(ParseError::NonConstantDefaultValue(
```

Comment:
What about using error directly from `eval_constant`?  So let it be:
```rust
*default_value = match eval_constant(working_set, &expression) {
    Ok(constant) => Some(constant),
    Err(e) => { working_set.error(e); None }
}
```

Reply:
So no need to change according to discussion:
https://github.com/nushell/nushell/pull/8940#discussion_r1172379102